src/System.CommandLine/Binding/ArgumentConverter.cs

```diff
@@ -330,17 +330,68 @@ public static bool CanBeBoundFromScalarValue(this Type type)
         [return: MaybeNull]
         internal static T GetValueOrDefault<T>(this ArgumentConversionResult result)
         {
-            switch (result)
+            return result switch
             {
-                case SuccessfulArgumentConversionResult successful:
-                    return (T)successful.Value!;
-                case FailedArgumentConversionResult failed:
-                    throw new InvalidOperationException(failed.ErrorMessage);
-                case NoArgumentConversionResult _:
-                    return default!;
-                default:
-                    return default!;
+                SuccessfulArgumentConversionResult successful => (T)successful.Value!,
+                FailedArgumentConversionResult failed => throw new InvalidOperationException(failed.ErrorMessage),
+                NoArgumentConversionResult _ => default!,
+                _ => default!,
+            };
+        }
+
+        [return: MaybeNull]
+        internal static T GetDefaultValue<T>()
+        {
+            return (T)GetDefaultValue(typeof(T));
+        }
+
+        private static MethodInfo EnumerableEmptyMethod { get; }
+            = typeof(Enumerable).GetMethod(nameof(Enumerable.Empty));
+
+        internal static object? GetDefaultValue(Type type)
+        {
+            if (type == typeof(string)) return "";
+            if (GetItemTypeIfEnumerable(type) is Type itemType)
+            {
+                if (type.IsArray)
+                {
+                    return CreateEmptyArray(itemType);
+                }
+                if (type.IsGenericType)
+                {
+                    return type.GetGenericTypeDefinition() switch
+                    {
+                        Type enumerable when enumerable == typeof(IEnumerable<>) => GetEmptyEnumerable(itemType),
+                        Type list when list == typeof(List<>) => GetEmptyList(itemType),
+                        Type array when array == typeof(IList<>) || 
+                                        array == typeof(ICollection<>) => CreateEmptyArray(itemType),
+                        _ => null
+                    };
+                }
             }
+            return type switch
+            {
+                Type nonGeneric 
+                    when nonGeneric == typeof(IList) ||
+                         nonGeneric == typeof(ICollection) ||
+                         nonGeneric == typeof(IEnumerable)
+                    => CreateEmptyArray(typeof(object)),
+                _ => null
+            };
+            
+            static object GetEmptyList(Type itemType)
+            {
+                return Activator.CreateInstance(typeof(List<>).MakeGenericType(itemType));
+            }
+
+            static IEnumerable GetEmptyEnumerable(Type itemType)
+            {
+                var genericMethod = EnumerableEmptyMethod.MakeGenericMethod(itemType);
```

Comment:
I wonder if these are worth caching.

Reply:
I could certainly put that in. Figured it was better to wait for perf benchmarks. But my suspicion is that caching may not offer much.

Reply:
Probably true.